lib/src/validator/relative_version_numbering.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:pub_semver/pub_semver.dart';
+
+import '../entrypoint.dart';
+import '../exceptions.dart';
+import '../package_name.dart';
+import '../pubspec.dart';
+import '../validator.dart';
+
+/// Gives a warning when publishing a new version, if the latest published
+/// version lower to this was not opted into null-safety.
+class RelativeVersionNumberingValidator extends Validator {
+  static const String guideUrl =
+      'http://dart.dev/null-safety-package-migration-guide';
```

Comment:
We need to check this is correct, right?

Also `https://`

Reply:
How about:

https://dart.dev/null-safety/migration-guide

Reply:
@kwalrath, up to you how you want to organize content on `dart.dev`. We can just setup a page that redirects, it's just important that the links we hard code here remains stable.

Looking at the current site maybe it could be `https://dart.dev/guides/libraries/null-safety` or `https://dart.dev/guides/null-safety-migration`.

Reply:
I was considering null safety to be along the lines of Dart 2, which has a homepage at https://dart.dev/dart-2. But obvs, a redirect is always possible. I'd rather not stick this under /guides.

Reply:
Changed to https://dart.dev/null-safety/migration-guide